django/db/models/expressions.py

```diff
@@ -925,6 +925,32 @@ def relabeled_clone(self, relabels):
         return self
 
 
+class Excluded(Expression):
+    def __init__(self, field_name):
+        super().__init__()
+        self.field_name = field_name
+
+    def resolve_expression(self, *args, **kwargs):
+        resolved = F(self.field_name).resolve_expression(*args, **kwargs)
+        self.output_field = resolved.output_field
+        return self
+
+    def as_sql(self, compiler, connection):
+        quoted_field_name = compiler.quote_name_unless_alias(self.field_name)
+        if connection.vendor in ["postgresql", "sqlite"]:
+            sql = f"EXCLUDED.{quoted_field_name}"
+        else:
+            raise NotImplementedError(
```

Comment:
```suggestion
            raise NotSupportedError(
```

Reply:
Also, this is untested.

Reply:
> Also, this is untested.

I don't understand. What hasn't been tested?

Reply:
I'm not sure how to say it clearer :thinking: This line is untested. Raised error is untested.

Reply:
i've rethought this approach. i'll update within 24 hours. i hope you enjoy.

Reply:
There is nothing to reconsider. Tests are **required** for new lines.

Reply:
> There is nothing to reconsider. Tests are **required** for new lines.

I'll make a note.

Reply:
Hi @felixxm, I'm looking for a way to reproduce exactly the local production environment for test execution. Do you have a suggestion?